app/actions/remote/entry/app.ts

```diff
@@ -53,15 +56,26 @@ export async function appEntry(serverUrl: string, since = 0, isUpgrade = false)
         }
     }
 
-    let switchToChannel = false;
+    const currentTeamIdAfterLoad = await getCurrentTeamId(database);
+    const currentChannelIdAfterLoad = await getCurrentChannelId(database);
 
-    // Immediately set the new team as the current team in the database so that the UI
-    // renders the correct team.
-    if (tabletDevice && initialChannelId) {
-        switchToChannel = true;
-        switchToChannelById(serverUrl, initialChannelId, initialTeamId);
-    } else {
-        setCurrentTeamAndChannelId(operator, initialTeamId, initialChannelId);
+    if (currentTeamIdAfterLoad !== currentTeamId) {
+        // Switched teams while loading
+        if (!(teamData.teams?.find((t) => t.id === currentTeamIdAfterLoad))) {
+            handleKickFromTeam(serverUrl, currentTeamIdAfterLoad);
+        }
+    } else if (currentTeamId !== initialTeamId) {
+        handleKickFromTeam(serverUrl, currentTeamId);
+    } else if (currentChannelIdAfterLoad !== currentChannelId) {
+        // Switched channels while loading
+        if (!(chData?.memberships?.find((m) => m.channel_id === currentChannelIdAfterLoad))) {
+            handleKickFromChannel(serverUrl, currentChannelIdAfterLoad);
+        }
+    } else if (currentChannelId !== initialChannelId) {
+        const navComponents = NavigationStore.getNavigationComponents();
+        if (tabletDevice || navComponents.includes(Screens.CHANNEL) || navComponents.includes(Screens.THREAD)) {
+            handleKickFromChannel(serverUrl, currentChannelId);
+        }
```

Comment:
If you switch teams / channels, the `currentTeamIdAfterLoad` will be different from the value we got before (`currentTeamId`). In any of those cases, it just checks that that you are a member or kick you otherwise. If you are still on the same channel, kick you if the entry logic is telling you to go to a different channel (you are no longer a member there).

When you moved to a new server, did you move all the way into the new server? Or where you still in the "create a new server/login" screen? In neither of those scenarios should happen, since you are not in the channel screen nor in the same server. Something I did not account for is moving to a channel, and going back. i.e. I move to channel b, go back to the channel list, this logic kicks in, I am no longer a member of channel b, this logic triggers the alert.

Finally, regarding the unreads, this logic still does not takes care of that. Probably will leave it for a different PR, since this one is complex enough (and I need still to handle the notifications route).